testsuite/tests/cxx-api/cxx-sarif.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/sh
+
+if command -v cygpath >/dev/null 2>&1; then
+  ocamltest_response="$(cygpath -u "${ocamltest_response:?}")"
+  have_cygpath=true
+else
+  have_cygpath=false
+fi
+
+if [ "${ccomp_type:?}" = "cc" ]; then
+  if "$have_cygpath"; then
+    ocamlsrcdir="$(cygpath -u "${ocamlsrcdir:?}")"
+    test_source_directory="$(cygpath -u "${test_source_directory:?}")"
+  fi
+
+  # TODO: use GCC 15 -fdiagnostics-add-output for stderr and SARIF output sinks
+  if ${cxx:?} -Wall -Wextra -Wpedantic \
+           -fdiagnostics-format=sarif-stderr \
+           ${cppflags:?} ${cflags:?} \
+           -I "${ocamlsrcdir:?}"/runtime \
+           -I "$ocamlsrcdir"/otherlibs/runtime_events \
+           -I "$ocamlsrcdir"/otherlibs/str \
+           -I "$ocamlsrcdir"/otherlibs/systhreads \
+           -I "$ocamlsrcdir"/otherlibs/unix \
+           -o stubs.o \
+           -c "${test_source_directory:?}"/stubs.cpp \
+           2> stubs.sarif; then
+    exit "$TEST_PASS"
+  else
+    cat <<'EOF' > filter.jq
+.runs.[0].results.[]
+| select(.message.text
+        | contains("ISO C++ forbids flexible array member") | not)
+| length
+EOF
+    length=$(jq -f filter.jq stubs.sarif)
+    if [ -z "$length" ]; then
+      exit "$TEST_PASS"
+    fi
+  fi
```

Comment:
Big comment here please as to what this is doing and why we skip that error. IIUC the point is that we get no warnings in the diagnostics apart from the ISO C++ one which we don't care about?

Is that string standardised, or is that sadly the best? From a distance, it's strange to have structured diagnostics, and then have to match a sentence!